src/components/dialogs/contingency-list/criteria-based/criteria-based-form.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { filterEquipmentDefinition } from '../../../../utils/equipment-types';
+import {
+    COUNTRIES_1,
+    COUNTRIES_2,
+    EQUIPMENT_TYPE,
+    NOMINAL_VOLTAGE_1,
+    NOMINAL_VOLTAGE_2,
+} from '../../../utils/field-constants';
+import React from 'react';
+import { useWatch } from 'react-hook-form';
+import { gridItem } from '../../../utils/dialog-utils';
+import { Grid } from '@mui/material';
+import CountriesInput from '../../../utils/rhf-inputs/countries-input';
+import RangeInput, {
+    getRangeInputSchema,
+} from '../../../utils/rhf-inputs/range-input';
+import yup from '../../../utils/yup-config';
+import SelectInput from '../../../utils/rhf-inputs/select-input';
+
+export const getCriteriaBasedSchema = () => ({
+    [EQUIPMENT_TYPE]: yup.string().nullable(),
+    [COUNTRIES_1]: yup.array().of(yup.string().nullable()),
+    [COUNTRIES_2]: yup.array().of(yup.string().nullable()),
+    ...getRangeInputSchema(NOMINAL_VOLTAGE_1),
+    ...getRangeInputSchema(NOMINAL_VOLTAGE_2),
+});
+
+const CriteriaBasedForm = ({ equipmentsTypes }) => {
```

Comment:
I think we could get equipmentTypes with Object.values(CONTINGENCY_LIST_EQUIPMENTS) instead

Reply:
I changed this component to be usable by contingency list and filter